matplotlib/mpl_plotting/mpl_plotting.py

```diff
@@ -53,32 +54,53 @@
 #    triangle markers.
 
 plt.plot(y1_noisy,'k-^')
+
 # 4. Overwrite the top left subplot with a plot of y1_noisy using a line plot
 #    with a red dashed line.
 plt.cla()
 plt.plot(y1_noisy,'r--')
+plt.title('Sine Wave')
+plt.ylabel('y1_noisy')
 # 5. In the top, middle subplot, plot y2_noisy using a line plot with
 #    green circle markers.
 plt.subplot(2,3,2)
 plt.plot(y2_noisy,'go')
+plt.title('Offset Sine')
+plt.ylabel('y2_noisy')
 # 6. In the top, right subplot, plot y1_noisy vs. y2_noisy using a scatter plot
 #    with blue circles.
 plt.subplot(2,3,3)
 plt.scatter(y1_noisy,y2_noisy,c='b')
+plt.title('Sine vs. Offset Sine')
+plt.xlabel('y1_noisy')
+plt.ylabel('y2_noisy')
 # 7. In the bottom, left subplot, plot the histogram of (y2_noisy - y1_noisy)
 plt.subplot(2,3,4)
 plt.hist(y2_noisy-y1_noisy)
+plt.title('Sine - Offset Sine')
+plt.ylabel('y2_noisy-y1_noisy')
+
 # 8. In the bottom, middle subplot, plot the histograms side by side for
 #    y1_noisy and y2_noisy (set hold to True again first).
 plt.subplot(2,3,5)
-plt.hist([y1_noisy,y2_noisy])
+plt.hist([y1_noisy,y2_noisy], label=['y1_noisy','y2_noisy'])
+plt.title('Sine & Offset Sine')
+plt.legend()
 # 9. In the bottom, right subplot, plot the correlation matrix of
 #        x1, y1, e1, y1_noisy, x2, y2, e2, y2_noisy
 #    as an image.
 #    Hint: syntax for correlation matrix is
 #        result = np.corrcoef([arr1, arr2, arr3, ...])
 plt.subplot(2,3,6)
-matrix = np.corrcoef([x1,y1,e1,y1_noisy,x2,y2,e2,y2_noisy])
+matrix = np.corrcoef([x1,y1,e1,y1_noisy,x2,y2,e2,y2_noisy] )
+labels=['x1','y1','e1','y1_noisy','x2','y2','e2','y2_noisy']
+x=[1,2,3,4,5,6,7]
+
+mask = np.zeros_like(matrix)
+
+w=seaborn.heatmap(matrix, cmap='viridis', vmax=1.0, vmin=-1.0 , mask = mask, linewidths=2.5)
+w.set_xticklabels(labels,rotation=45)
+w.set_yticklabels(labels[::-1],rotation=45)
```

Comment:
I feel like there might be a more efficient way of adding labels to a correlation matrix.

Reply:
There is a package called `pandas` that makes this kind of thing much easier.  For now, this is fine, and the "pain" of creating the labels manually demonstrates the appeal of a package like Pandas.